definitions.h

```diff
@@ -47,6 +52,327 @@ struct Vertex
+		Matrix operator*(const Matrix & m) const;
+		//Matrix operator*(const Vertex & m) const;
+		Vertex operator*(const Vertex & m) const;
+		Matrix& operator=(Matrix const& m);
+		Matrix & operator=(std::initializer_list<double> m);
+		Matrix & operator=(std::initializer_list<std::initializer_list<double>> m);
+
+    private:
+        int column;
+        int row;
+		int size;
+        double matrixData[16];
+};
+Matrix::Matrix()
+{
+	column = 0;
+    row = 0;
+	size = 0;
+}
+
+Matrix::Matrix (int c, int r)
+{
+    column = c;
+    row = r;
+	size = row * column;
+	for(int i = 0; i < size; i++)
+	{
+		matrixData[i] = 0;
+	}
+}
+
+void Matrix::insert(int x, int y, double data)
+{
+    matrixData[(x * column) + y] = data;
+}
+
+double Matrix::returnVal(int x, int y)
+{
+    return matrixData[(x * column) + y];
+}
+
+int Matrix::getColumnSize()
+{
+	return column;
+}
+
+int Matrix::getRowSize()
+{
+	return row;
+}
+
+int Matrix::getSize()
+{
+	return size;
+}
+
+Matrix computeDimensions(int c1, int r1, int c2, int r2)
+{
+	if (c1 < c2 && r1 == r2)
+	{
+		Matrix result = Matrix(c1, r1);
+		return result;
+	}
+	else if (c1 == c2 && r1 < r2)
+	{
+		Matrix result = Matrix(c1, r1);
+		return result;
+	}
+
+	else if (c2 < c1 && r2 == r1)
+	{
+		Matrix result = Matrix(c2, r2);
+		return result;
+	}
+	else if (c2 == c1 && r2 < r1)
+	{
+		Matrix result = Matrix(c2, r2);
+		return result;
+	}
+	else if (c1 == c2 && r1 == r2)
+	{
+		Matrix result = Matrix(c1, r1);
+		return result;
+	}
+	else
+	{
+		std::cerr << "WARNING: DIMENSIONAL MISMATCH ABORTING PROGRAM\n";
+		exit(1);
+	}
+}
+
+/********************************************************
+* Matrix Copy
+*
+* Overloading the equal operator doesn't work I don't
+* know why so I'm making this.
+********************************************************/
+/*
+Matrix Matrix::matrixCopy(Matrix & m)
+{
+	delete [] matrixData;
+	matrixData = NULL;
+	this->~Matrix();
+	
+	
+	column = m.column;
+	row = m.row;
+	size = m.size;
+	
+	matrixData = new double[size];
+	
+	for (int i = 0; i < size; i++)
+	{
+		//cout << "entering in " << m.matrixData[i] << " at " << i << endl;
+		matrixData[i] =  m.matrixData[i];
+		//cout << "matrixData[" << i << "] is " << this->matrixData[i] << endl;
+	}
+	
+	
+	
+	return *this;
+}
+*/
+
+
+/******************************************************
+* * Multiplication Operator Overload
+*
+* 
+******************************************************/
+Matrix Matrix::operator*(const Matrix & m) const
+{
+	Matrix result = computeDimensions(column, row, m.column, m.row);
+	//Matrix result = Matrix(1, 3);
+
+	/*result.insert(0, 0, 5);
+	cout << "inserted 5\n";
+	result.insert(0, 1, 5);
+	cout << "inserted 5\n";
+	result.insert(0, 2, 6);*/
+
+
+	for (int x = 0; x < row; x++)
+	{
+		for (int y = 0; y < m.column; y++)
+		{
+			for (int k = 0; k < column; k++)
+			{
+				result.insert(x, y, (result.returnVal(x, y) + (matrixData[(x * column)+k] * m.matrixData[(k * m.column) + y])));

```

Comment:
Try adding comments here to explain how you're handling the matrix with a 1D array.